parser.y

```diff
@@ -242,6 +242,7 @@ import (
 	sqlSmallResult    "SQL_SMALL_RESULT"
 	ssl               "SSL"
 	starting          "STARTING"
+	statistics        "STATISTICS"
```

Comment:
This would make `statistics` a reserved keyword, while in previous syntax implementation, it is just a TiDB keyword. If we don't do this, the newly added rule would have shift/reduce conflict with `ColumnKeywordOpt`, because `statistics` can be treated as a column name for `ADD [COLUMN] statistics`.

Reply:
Making `statistics` or `stats` a reserved keyword just to avoid having the "frankenstein" `CREATE STATISTICS` syntax (already used by T-SQL and PostgreSQL) is not a good trade-off imo.

Reply:
@zz-jason @morgo what's your opinion?

Reply:
For better syntax compatibility with MySQL, I think it's worthy.

Reply:
and what about `TIDB_STATS` / `TIDB_STATISTICS` 🤔

Reply:
I agree with kennytm. `stats` / `statistics` is not a reserved word in the SQL standard, or in MySQL. It doesn't hurt usability too much to use `TIDB_STATISTICS` or `OPTIMIZER_STATISTICS` instead.

Reply:
Alright, I replaced `statistics` with `tidb_stats`, shall we use `admin reload statistics` or `admin reload tidb_stats` then?

Reply:
I like the consistency of `admin reload tidb_stats`, but not a strong opinion.

Reply:
If we choose `tidb_stats`, lots of SQL grammar might need to be modified to keep consistency. 

For example, some SQL syntaxes:

```sql
show stats_meta;
load stats;
```

Config:

```
stats-lease = "3s"
```

HTTP API:

```
http://$host:$port/stats/dump/$db/$table
```

And some tables:

```
stats_buckets
stats_feedback
stats_histograms
stats_meta
stats_top_n
```

keeping the name to be consistent helps users to easily remember these concepts. I prefer to use `stats` for drop and alter table statements.

Reply:
What about replacing `TIDB_STATS` with `STATS_EXTENDED` to keep consistency? @zz-jason 
```
ALTER TABLE t ADD stats_extended s1 dependency(a,b);
ALTER TABLE t DROP stats_extended s1;
ADMIN RELOAD stats_extended;
```
or maybe a shorter `STATS_EXT`?